File: benchmarks/somadigitos.c
#include <stdio.h>

int somaDigitos(int n) {
    int soma = 0;
    while (n > 0) {
        soma += n % 10;
        n /= 10;
    }
    return soma;
}

int main() {
    int total = 0;
    for (int i = 1; i <= 10000000; i++) {
        total += somaDigitos(i);
    }
    printf("%d\n", total);
    return 0;
}

File: benchmarks/somadigitos.rs
fn soma_digitos(mut n: i32) -> i32 {
    let mut soma = 0;
    while n > 0 {
        soma += n % 10;
        n /= 10;
    }
    soma
}

fn main() {
    let mut total: i32 = 0;
    for i in 1..=10_000_000 {
        total += soma_digitos(i);
    }
    println!("{}", total);
}
